contracts/pcv/utils/NamedStaticPCVDepositWrapper.sol

```diff
@@ -0,0 +1,180 @@
+  @author Fei Protocol
+
+  Returns PCV in USD terms
+*/
+contract NamedStaticPCVDepositWrapper is IPCVDepositBalances, CoreRef {
+    using SafeCast for *;
+
+    // -------------- Events ---------------
+    event BalanceUpdate(uint256 oldBalance, uint256 newBalance, uint256 oldFEIBalance, uint256 newFEIBalance);
+
+    event DepositRemoved(uint256 index);
+
+    /// @notice struct to store info on each PCV Deposit
+    struct PCVDeposit {
+        string depositName;
+        uint256 usdAmount; /// USD equivalent in this deposit, not including FEI value
+        uint256 feiAmount; /// amount of FEI in this deposit
+        uint256 underlyingTokenAmount; /// amount of underlying token in this deposit
+        address underlying; /// address of the underlying token this deposit is reporting
+    }
+
+    /// @notice a list of all pcv deposits
+    PCVDeposit[] public pcvDeposits;
+
+    /// @notice the PCV balance
+    uint256 public override balance;
+
+    /// @notice the reported FEI balance
+    uint256 public feiReportBalance;
+
+    constructor(address _core, PCVDeposit[] memory newPCVDeposits) CoreRef(_core) {
+
+        // Uses oracle admin to share admin with CR oracle where this contract is used
+        _setContractAdminRole(keccak256("ORACLE_ADMIN_ROLE"));
+
+        // add all pcv deposits
+        for (uint256 i = 0; i < newPCVDeposits.length; i++) {
+            _addDeposit(newPCVDeposits[i]);
+        }
+    }
+
+    /// @notice helper method to delete a PCV deposit
+    function _removeDeposit(uint256 index) internal {
+        PCVDeposit storage removePCVDeposit = pcvDeposits[index];
+
+        uint256 depositBalance = removePCVDeposit.usdAmount;
+        uint256 feiDepositBalance = removePCVDeposit.feiAmount;
+
+        balance -= depositBalance;
+        feiReportBalance -= feiDepositBalance;
+        delete pcvDeposits[index];
+
+        emit BalanceUpdate(balance + depositBalance, balance, feiReportBalance + feiDepositBalance, feiReportBalance);
+        emit DepositRemoved(index);
+    }
+
+    /// @notice helper method to add a PCV deposit
+    function _addDeposit(PCVDeposit memory newPCVDeposit) internal {
+        require(newPCVDeposit.feiAmount > 0 || newPCVDeposit.usdAmount > 0, "NamedStaticPCVDepositWrapper: must supply either fei or usd amount");
+
+        uint256 oldBalance = balance;
+        uint256 oldFEIBalance = feiReportBalance;
+
+        balance += newPCVDeposit.usdAmount;
+        feiReportBalance += newPCVDeposit.feiAmount;
+        pcvDeposits.push(newPCVDeposit);
+
+        emit BalanceUpdate(oldBalance, balance, oldFEIBalance, feiReportBalance);
+    }
+
+    /// @notice helper method to edit a PCV deposit
+    function _editDeposit(
+        uint256 index,
+        uint256 usdAmount,
+        uint256 feiAmount,
+        uint256 underlyingTokenAmount,
+        string calldata depositName,
+        address underlying
```

Comment:
could you keep the same order than in the structure ? this way, all functions will take arguments in a coherent order

Reply:
fixed